python-runtime/ftl/src/ftl/extract/context.py

```diff
@@ -0,0 +1,95 @@
+import importlib.util
+import multiprocessing
+import threading
+
+from ftl.protos.xyz.block.ftl.v1.schema import schema_pb2 as schemapb
+
+"""Global context across multiple files in a package."""
+
+
+class GlobalExtractionContext:
+    def __init__(self):
+        manager = multiprocessing.Manager()
+        self.needs_extraction = manager.dict()
+        self.verbs = manager.dict()
+        self.data = manager.dict()
+
+    def deserialize(self):
+        deserialized_decls = {}
+        for ref_key, serialized_decl in self.verbs.items():
+            decl = schemapb.Verb()
+            decl.ParseFromString(serialized_decl)
+            deserialized_decls[ref_key] = decl
+        for ref_key, serialized_decl in self.data.items():
+            decl = schemapb.Data()
+            decl.ParseFromString(serialized_decl)
+            deserialized_decls[ref_key] = decl
+        return deserialized_decls
+
+    def init_local_context(self):
+        return LocalExtractionContext(self.needs_extraction, self.verbs, self.data)
+
+
+"""Local context for a single Python file."""
```

Comment:
Is this meant to be attached to the class below?

Reply:
aye done